exporter/elasticsearchexporter/internal/serializer/otelserializer/serializeprofiles/transform.go

```diff
@@ -333,9 +334,20 @@ func stackFrames(dic pprofile.ProfilesDictionary, profile pprofile.Profile, samp
 func getFrameID(dic pprofile.ProfilesDictionary, location pprofile.Location) (*libpf.FrameID, error) {
 	// The MappingIndex is known to be valid.
 	mapping := dic.MappingTable().At(int(location.MappingIndex()))
-	buildID, err := getBuildID(dic, mapping)
-	if err != nil {
-		return nil, err
+	fileID, err := getBuildID(dic, mapping)
+	if err != nil || fileID.IsZero() {
+		// Create an artificial file ID if the build ID is not available.
```

Comment:
Nit (all file IDs are artificial):

```suggestion
		// Synthesize a file ID if the build ID is not available.
```